fluss-common/src/main/java/com/alibaba/fluss/row/encode/iceberg/IcebergBinaryRowWriter.java

```diff
@@ -0,0 +1,326 @@
+        ensureCapacity(8);
+        UnsafeUtils.putDouble(buffer, cursor, value);
+        cursor += 8;
+    }
+
+    public void writeString(BinaryString value) {
+        // Based on Iceberg's Conversions.toByteBuffer() for STRING type:
+        // Strings are encoded as UTF-8 bytes without length prefix
+        // Reference:https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/types/Conversions.java
+        byte[] bytes = BinaryString.encodeUTF8(value.toString());
+        ensureCapacity(bytes.length);
+        segment.put(cursor, bytes, 0, bytes.length);
+        cursor += bytes.length;
+    }
+
+    void writeBytes(byte[] bytes) {
+        // Based on Iceberg's Conversions.toByteBuffer() for BINARY type:
+        // Binary data is stored directly without length prefix
+        ensureCapacity(bytes.length);
+        segment.put(cursor, bytes, 0, bytes.length);
+        cursor += bytes.length;
+    }
+
+    public void writeDecimal(Decimal value, int precision) {
+        // Iceberg stores decimals as unscaled big-endian byte arrays
+        // Reference:https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/types/Conversions.java
+        byte[] unscaledBytes = value.toUnscaledBytes();
+
+        // For Iceberg, decimals are stored with fixed byte lengths based on precision
+        int requiredBytes = getIcebergDecimalBytes(precision);
+
+        byte[] icebergBytes = new byte[requiredBytes];
+
+        // Convert to big-endian format with proper padding
+        if (unscaledBytes.length <= requiredBytes) {
+            // Pad with sign extension
+            byte paddingByte =
+                    (unscaledBytes.length > 0 && (unscaledBytes[0] & 0x80) != 0)
+                            ? (byte) 0xFF
+                            : (byte) 0x00;
+
+            Arrays.fill(icebergBytes, 0, requiredBytes - unscaledBytes.length, paddingByte);
+            System.arraycopy(
+                    unscaledBytes,
+                    0,
+                    icebergBytes,
+                    requiredBytes - unscaledBytes.length,
+                    unscaledBytes.length);
+        } else {
+            // Truncate if too large (should not happen with proper validation)
+            System.arraycopy(unscaledBytes, 0, icebergBytes, 0, requiredBytes);
+        }
+
+        writeBytes(icebergBytes);
+    }
+
+    public void writeTimestampNtz(TimestampNtz value, int precision) {
+        // Iceberg stores timestamps as microseconds since epoch
+        // Reference:
+        // https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/types/Conversions.java
+        long micros = value.getMillisecond() * 1000L + (value.getNanoOfMillisecond() / 1000L);
+        writeLong(micros);
+    }
+
+    public void writeTimestampLtz(TimestampLtz value, int precision) {
+        // Iceberg stores timestamptz as microseconds since epoch in UTC
+        // Reference:
```

Comment:
We don't need the Reference  comment for every method since we have commented it in header